services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java

```diff
@@ -164,7 +166,18 @@ public void persistDiagnosisKeysPayload(SubmissionPayload protoBufDiagnosisKeys)
     diagnosisKeyService.saveDiagnosisKeys(diagnosisKeys);
   }
 
-  private synchronized void updateFakeDelay(long realRequestDuration) {
-    fakeDelay = fakeDelay + (1 / fakeDelayMovingAverageSamples) * (realRequestDuration - fakeDelay);
+  /** @implNote VisibleForTesting */
+  void updateFakeDelay(long realRequestDuration) {
+    while (true) {
+      Double fd = fakeDelay.get();
+      if (fakeDelay.compareAndSet(
+          fd, fd + (1 / fakeDelayMovingAverageSamples) * (realRequestDuration - fd))) {
+        break;
```

Comment:
`while (true) .. break` is not a good code style.

For the code it seems irrelevant which thread changes fakeDelay at which point (something this code does not fix either), therefore compareAndSet is unnecessary. The only relevant thing is to make sure that the first fakeDelay value does not differ from the second fakeDelay value in the above calculate. And that a simple final value would do, using an Atomic in this case is overkill.